sky/serve/load_balancer.py

```diff
@@ -180,11 +191,41 @@ async def _proxy_with_retries(
         # SkyServe supports serving on Spot Instances. To avoid preemptions
         # during request handling, we add a retry here.
         retry_cnt = 0
+        # We keep track of the failed replicas for the current request, because
+        # we have a global round-robin policy, and if there is a large load,
+        # all retries for the same request can go to the same replica by chance.
+        # If the same replica is in `NOT_READY` state but the new state has not
+        # been synced from the controller, the current request will fail.
+        #
+        # We maintain a per-request failed replica set instead of the global
+        # one to allow multiple requests to still try failed replicas for one
+        # time, in case that replica is failed by transient network issue.
+        failed_replica_urls: Set[str] = set()
         while True:
             retry_cnt += 1
             with self._client_pool_lock:
+
+                def _should_reset_failed_replicas_for_retry(
+                    failed_replica_urls: Set[str],) -> bool:
```

Comment:
```suggestion
                    failed_replica_urls: Set[str]) -> bool:
```
nit. also, this argument is shadowing the outer variable name. is it intended?